clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -142,92 +167,142 @@ public PrototypeAsyncConsumer(final Time time,
-                applicationEventProcessorSupplier,
-                networkClientDelegateSupplier,
-                requestManagersSupplier);
-
-        // These are specific to the foreground thread
-        FetchConfig<K, V> fetchConfig = createFetchConfig(config, deserializers);
-        this.fetchBuffer = new FetchBuffer<>(logContext);
-        this.fetchCollector = new FetchCollector<>(logContext,
-                metadata,
-                subscriptions,
-                fetchConfig,
-                fetchMetricsManager,
-                time);
+        try {
+            GroupRebalanceConfig groupRebalanceConfig = new GroupRebalanceConfig(config,
+                    GroupRebalanceConfig.ProtocolType.CONSUMER);
+
+            this.groupId = Optional.ofNullable(groupRebalanceConfig.groupId);
+            this.clientId = config.getString(CommonClientConfigs.CLIENT_ID_CONFIG);
+            LogContext logContext = createLogContext(config, groupRebalanceConfig);
+            this.log = logContext.logger(getClass());
+            groupId.ifPresent(groupIdStr -> {
+                if (groupIdStr.isEmpty()) {
+                    log.warn("Support for using the empty group id by consumers is deprecated and will be removed in the next major release.");
+                }
+            });
+
+            log.debug("Initializing the Kafka consumer");
+            this.requestTimeoutMs = config.getInt(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG);
+            this.defaultApiTimeoutMs = config.getInt(ConsumerConfig.DEFAULT_API_TIMEOUT_MS_CONFIG);
+            this.time = time;
+            this.metrics = createMetrics(config, time);
+            this.retryBackoffMs = config.getLong(ConsumerConfig.RETRY_BACKOFF_MS_CONFIG);
+
+            List<ConsumerInterceptor<K, V>> interceptorList = getConfiguredConsumerInterceptors(config);
+            this.interceptors = new ConsumerInterceptors<>(interceptorList);
+            this.deserializers = new Deserializers<>(config, keyDeserializer, valueDeserializer);
+            this.subscriptions = createSubscriptionState(config, logContext);
+            ClusterResourceListeners clusterResourceListeners = ClientUtils.configureClusterResourceListeners(metrics.reporters(),
+                    interceptorList,
+                    Arrays.asList(deserializers.keyDeserializer, deserializers.valueDeserializer));
+            this.metadata = new ConsumerMetadata(config, subscriptions, logContext, clusterResourceListeners);
+            final List<InetSocketAddress> addresses = ClientUtils.parseAndValidateAddresses(config);
+            metadata.bootstrap(addresses);
+
+            FetchMetricsManager fetchMetricsManager = createFetchMetricsManager(metrics);
+            this.isolationLevel = getConfiguredIsolationLevel(config);
+
+            ApiVersions apiVersions = new ApiVersions();
+            final BlockingQueue<ApplicationEvent> applicationEventQueue = new LinkedBlockingQueue<>();
```

Comment:
maybe add a note here about bounding the queues.  I think we will need to implement this.

Reply:
Do you mind if I create a Jira for this instead?